plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlValidatorTest.java

```diff
@@ -54,9 +56,12 @@
   public static void setUp() throws CoreException {
     project = projectCreator.getProject();
     createFolders(project, new Path("src/main/webapp/WEB-INF"));
-    IFile webXml = project.getFile("src/main/webapp/WEB-INF/web.xml");
-    webXml.create(new ByteArrayInputStream(new byte[0]), true, null);
-    resource = webXml;
+    resource = project.getFile("src/main/webapp/WEB-INF/web.xml");
+    resource.create(new ByteArrayInputStream(new byte[0]), true, null);
+    createFolders(project, new Path("src/main/java"));
+    IFile file = project.getFile("src/main/java/ServletClass.java");
+    file.create(new ByteArrayInputStream(
+        "public class ServletClass {}".getBytes(StandardCharsets.UTF_8)), true, null);
```

Comment:
We could have a separate check that the referenced class is actually a Servlet, though I'm not sure how often that comes up in practice.